modules/videopress/editor-media-view.php

```diff
@@ -0,0 +1,212 @@
+}
+
+/**
+ * Media List:
+ * Do the same as ^^ but for the list view.
+ */
+add_action( 'pre_get_posts', 'videopress_media_list_table_query' );
+function videopress_media_list_table_query( $query ) {
+	if ( is_admin() && $query->is_main_query() && ( 'upload' === get_current_screen()->id ) ) {
+		$meta_query = array(
+			array(
+				'key'     => 'videopress_poster_image',
+				'compare' => 'NOT EXISTS',
+			),
+		);
+
+		if ( $old_meta_query = $query->get( 'meta_query' ) ) {
+			$meta_query[] = $old_meta_query;
+		}
+
+		$query->set( 'meta_query', $meta_query );
+	}
+}
+
+/**
+ * Make sure that any Video that has a VideoPress GUID passes that data back.
+ */
+add_filter( 'wp_prepare_attachment_for_js', 'videopress_prepare_attachment_for_js' );
+function videopress_prepare_attachment_for_js( $post ) {
+	if ( 'video' === $post['type'] ) {
+		$guid = get_post_meta( $post['id'], 'videopress_guid' );
+		if ( $guid ) {
+			$post['videopress_guid'] = $guid;
+		}
+	}
+	return $post;
+}
+
+/**
+ * Wherever the Media Modal is deployed, also deploy our overrides.
+ */
+add_action( 'wp_enqueue_media', 'add_videopress_media_overrides' );
+function add_videopress_media_overrides() {
+	add_action( 'admin_print_footer_scripts', 'videopress_override_media_templates', 11 );
+}
+
+/**
+ * Our video overrides!
+ *
+ * We have a template for the iframe to get injected.
+ */
+function videopress_override_media_templates(){
+	?>
+	<script type="text/html" id="tmpl-videopress_iframe_vnext">
+		<iframe style="display: block; max-width: 100%;" width="{{ data.width }}" height="{{ data.height }}" src="https://videopress.com/embed/{{ data.guid }}?{{ data.urlargs }}" frameborder='0' allowfullscreen></iframe>
+	</script>
+	<script>
+		(function( media ){
+			// This handles the media library modal attachment details display.
+			if ( typeof media.view.Attachment.Details.TwoColumn !== 'undefined' ) {
+				var TwoColumn   = media.view.Attachment.Details.TwoColumn,
+					old_render  = TwoColumn.prototype.render,
+					vp_template = wp.template('videopress_iframe_vnext');
+
+				TwoColumn.prototype.render = function () {
+					// Have the original renderer run first.
+					old_render.apply(this, arguments);
+
+					// Now our stuff!
+					if ('video' === this.model.get('type')) {
+						if (this.model.get('videopress_guid')) {
+							this.$('.attachment-media-view .thumbnail-video').html(vp_template({
+								guid: this.model.get('videopress_guid'),
+								width: this.model.get('width'),
+								height: this.model.get('height')
+							}));
+						}
+					}
+				};
+			} else { console.log( 'media.view.Attachment.Details.TwoColumn undefined' ); }
+
+			// This handles the recreating of the shortcode when editing the mce embed.
+			if ( typeof media.video !== 'undefined' ) {
+				var old_video_shortcode = media.video.shortcode;
+
+				media.video.defaults.videopress_guid = '';
+				media.video.shortcode = function (model) {
+					model.videopress_guid = 'FOOBAR';
```

Comment:
Why was this foobar?  Just a dummy guid at the moment.  Needs to be redone properly.